### PR TITLE
fix: fixed error on submission of sales order to send email for delivery window time

### PR DESCRIPTION
**Pull request by MyuddinKhatri**
Ref: [Asana](https://app.asana.com/0/1192407896547052/1200233350242169)

### CODE REVIEW

File: bloomstack_core/hook_events/utils.py [hunk 1]
```diff
@@ -85,3 +84,12 @@ def validate_delivery_window(doc, method):
...
 			frappe.sendmail(recipients=recipients, subject=subject, message=message)
+
+def get_users_with_role(role):
+	# returns users with the specified role
+	return [user for users in frappe.db.sql("""SELECT DISTINCT `tabUser`.`name`
+		FROM `tabHas Role`, `tabUser`
+		WHERE `tabHas Role`.`role`=%s
+		AND `tabUser`.`name`!='Administrator'
+		AND `tabHas Role`.`parent`=`tabUser`.`name`
+		AND `tabUser`.`enabled`=1""", role) for user in users]
```
Comment (hrwX) [inline bloomstack_core/hook_events/utils.py]: - Use ORM

**Review by hrwX — CHANGES_REQUESTED** [review]

File: bloomstack_core/hook_events/utils.py [hunk 2]
```diff
@@ -85,3 +84,16 @@ def validate_delivery_window(doc, method):
 				))
 
 			frappe.sendmail(recipients=recipients, subject=subject, message=message)
+
+def get_users_with_role(role):
+	# returns users with the specified role
+	user_list = frappe.get_all("User", fields="name",
```
Comment (hrwX) [inline bloomstack_core/hook_events/utils.py]:
```suggestion
	user_list = frappe.get_all("User", fields=["`tabUser`.name"],
```

File: bloomstack_core/hook_events/utils.py [hunk 3]
```diff
@@ -85,3 +84,16 @@ def validate_delivery_window(doc, method):
...
+		filters = [
+			["Has Role", "role", "=", role],
+			["User", "name", "!=", "Administrator"],
+			["User", "enabled", "=", 1]
+		],
+		as_list=1
+	)
+
+	return[user for users in user_list for user in users]
```
Comment (hrwX) [inline bloomstack_core/hook_events/utils.py]:
```suggestion
	return [user for users in user_list for user in users]
```

**Review by hrwX — CHANGES_REQUESTED** [review]

File: bloomstack_core/hook_events/utils.py [hunk 4]
```diff
@@ -85,3 +84,16 @@ def validate_delivery_window(doc, method):
 				))
 
 			frappe.sendmail(recipients=recipients, subject=subject, message=message)
+
+def get_users_with_role(role):
+	# returns users with the specified role
+	user_list = frappe.get_all("User", fields=["`tabUser`.name"],
+		filters = [
+			["Has Role", "role", "=", role],
+			["User", "name", "!=", "Administrator"],
```
Comment (hrwX) [inline bloomstack_core/hook_events/utils.py]: Shouldnt it be admin and guest ?

**Review by hrwX — CHANGES_REQUESTED** [review]

**Review by hrwX — APPROVED** [review]

**Review by sahil28297 — APPROVED** [review]

**Pull request merged**